src/components/timeRangePicker/TimeRangePicker.tsx

```diff
@@ -0,0 +1,302 @@
+        <Label name="Hours:" />
+        <UnifiedMultipleSelect size="small">
+          {DEFAULT_TIME_RANGE_OPTIONS.map((timeRangeOption) => (
+            <SelectButton
+              key={timeRangeOption.id}
+              name={timeRangeOption.id}
+              value={timeRangeOption.id}
+              labelText={`${timeRangeOption.start} - ${timeRangeOption.end}`}
+              onClick={() => setSelectedTimeRange(() => timeRangeOption)}
+              isActive={timeRangeOption.id === selectedTimeRange.id}
+            />
+          ))}
+          <SelectButton
+            key="custom"
+            name="custom"
+            value="custom"
+            labelText="custom"
+            onClick={() =>
+              setSelectedTimeRange((prev) => {
+                return {
+                  ...prev,
+                  id: "custom",
+                };
+              })
+            }
+            isActive={selectedTimeRange.id === "custom"}
+          />
+        </UnifiedMultipleSelect>
+        <CustomTimeRangeSelector isVisible={showCustomTimeRange}>
+          <div>From</div>
+          <input
+            type="time"
+            name="time-range-start"
+            value={selectedTimeRange.start}
+            onChange={(e) => {
+              e.persist();
+              setSelectedTimeRange((prev) => {
+                return {
+                  ...prev,
+                  start: e.target.value,
+                };
+              });
+            }}
+          />
+          <div>to</div>
+          <input
+            type="time"
+            name="time-range-end"
+            value={selectedTimeRange.end}
+            min={selectedTimeRange.start}
+            onChange={(e) => {
+              e.persist();
+              setSelectedTimeRange((prev) => {
+                return {
+                  ...prev,
+                  end: e.target.value,
+                };
+              });
+            }}
+          />
+          <Button severity="high" onClick={submit} size="sm">
+            Apply
+          </Button>
+        </CustomTimeRangeSelector>
+      </Row>
+    </>
+  );
+}
+
+const getMonday = (date: Date): Date => {
+  const result = new Date(date);
+
+  if (isMonday(result)) return result;
+  else return previousMonday(result);
+};
+
+const getInitSelectedDate = (initRange: Interval): Date => {
+  let initDate: Date;
+
+  if (isSameDay(initRange.start, initRange.end)) {
+    initDate = new Date(initRange.start);
+  } else {
+    initDate = new Date();
+  }
+
+  return initDate;
+};
+
+const getDateOptions = (monday: Date): DateOption[] => {
+  let result: DateOption[] = [];
+  let current = new Date(monday.getTime());
+
+  WEEKDAYS.forEach((weekday) => {
+    result.push({
+      id: current.toDateString(),
+      weekday: weekday,
+      day: current.getDate(),
+      month: current.getMonth() + 1,
+      year: current.getFullYear(),
+    });
+
+    current.setDate(current.getDate() + 1);
+  });
+
+  return result;
```

Comment:
I think we can utliize `map`

```suggestion
  const mondayTime = monday.getTime();
  
  return WEEKDAYS.map((day, index) => {
      const currentDate = new Date(mondayTime) + index; // this should probably be handled differently but i think you get the idea
      
    return {
      id: currentDate.toDateString(),
      weekday: weekday,
      day: currentDate.getDate(),
      month: currentDate.getMonth() + 1,
      year: currentDate.getFullYear(),
    }
  });
```

Reply:
@nowyDEV Hi Dominik, thanks for your suggestion! May I know what's the advantages by using `map`? 
Because I noticed that in the `map` method, we need to declare  for each item a new `currentDate`, and the forEach method needs only one.

Reply:
I thought it's easier to understand since everything is scoped inside the map callback, the `currentDate` is also declared before being used and not reassigned at the end of the function (forEach callback).